src/iceberg/v1_metadata.cc

```diff
@@ -68,9 +68,13 @@ Result<std::shared_ptr<StructType>> ManifestEntryAdapterV1::GetManifestEntryType
   // Deprecated. Always write a default in v1. Do not write in v2 or v3.
   static const SchemaField kBlockSizeInBytes = SchemaField::MakeRequired(
       105, "block_size_in_bytes", int64(), "Block size in bytes");
-  ICEBERG_ASSIGN_OR_RAISE(auto partition_type, partition_spec_->PartitionType());
+  std::shared_ptr<StructType> partition_type = nullptr;
+  if (partition_spec_ != nullptr && current_schema_ != nullptr) {
+    ICEBERG_ASSIGN_OR_RAISE(partition_type,
+                            partition_spec_->PartitionType(*current_schema_));
+  }
```

Comment:
```suggestion
    ICEBERG_ASSIGN_OR_RAISE(auto partition_type,
                            partition_spec_->PartitionType(*current_schema_));
```
This is wrong since `current_schema_` and `partition_spec_` cannot be null.